### PR TITLE
use secrets for asg

### PR DESCRIPTION
**Pull request by gourao**

### CODE REVIEW

File: _includes/k8s-spec-generate.md [hunk 1]
```diff
@@ -4,11 +4,40 @@ To generate the spec file for the 1.3 release, head on to [1.3 install page](htt
...
+Use `kubectl` to create the secret, for example:
+
+```
+# kubectl -n kube-system create secret generic etcd-certs --from-file=etcd-secrets/
+```
+
+Now edit the Portworx spec file to reference the certificates.  For example, assuming the names of the files are `pwx-ca.crt` `pwx-user-cert.crt` and `pwx-user-key.key`, modify the spec file as follows:
+
+```
+  - mountPath: /etc/pwx/etcdcerts
```
Comment (piyush-nimbalkar) [inline _includes/k8s-spec-generate.md]: let's add volumeMounts: at the top, so people know where to add this section
```
volumeMounts:
- mountPath: /etc/pwx/etcdcerts
   name: etcdcerts
```

File: _includes/k8s-spec-generate.md [hunk 2]
```diff
@@ -4,11 +4,40 @@ To generate the spec file for the 1.3 release, head on to [1.3 install page](htt
 
 Alternately, you can use curl to generate the spec as described in [Generating Portworx Kubernetes spec using curl](/scheduler/kubernetes/px-k8s-spec-curl.html).
 
->**Secure ETCD:**<br/> If using secure etcd provide "https" in the URL and make sure all the certificates are in the _/etc/pwx/_ directory on each host which is bind mounted inside PX container.
+#### Secure ETCD and Certificates
+If using secure etcd provide "https" in the URL and make sure all the certificates are in the _/etc/pwx/_ directory on each host which is bind mounted inside PX container.
 
+#### Using Secrets to Provision Certificates
+>**Note for ASG users:**<br/>When using PX with Auto Scaling Groups in AWS, you should use Secrets to have Kubernetes provision the ETCD certificates.  This way, the certificates will be provisioned on any new EC2 instance automatically.
```
Comment (harsh-px) [inline _includes/k8s-spec-generate.md]: This isn't specific to ASG. But rather can be done in any k8s cluster. So we can reword this to

```
It is recommended to use Kubernetes Secrets to provide ETCD certificates to Portworx. This way, the certificates will be automatically mounted when new nodes join the cluster."
```

File: _includes/k8s-spec-generate.md [hunk 3]
```diff
@@ -4,11 +4,40 @@ To generate the spec file for the 1.3 release, head on to [1.3 install page](htt
...
+
+Now edit the Portworx spec file to reference the certificates.  For example, assuming the names of the files are `pwx-ca.crt` `pwx-user-cert.crt` and `pwx-user-key.key`, modify the spec file as follows:
+
+```
+  - mountPath: /etc/pwx/etcdcerts
+    name: etcdcerts
+```
+
+```
+  - name: etcdcerts
```
Comment (piyush-nimbalkar) [inline _includes/k8s-spec-generate.md]:
```
volumes:
- secret:
  secretName: etcd-certs
   ....
```

File: _includes/k8s-spec-generate.md [hunk 4]
```diff
@@ -4,11 +4,40 @@ To generate the spec file for the 1.3 release, head on to [1.3 install page](htt
...
+If using secure etcd provide "https" in the URL and make sure all the certificates are in the _/etc/pwx/_ directory on each host which is bind mounted inside PX container.
 
+#### Using Secrets to Provision Certificates
+>**Note for ASG users:**<br/>When using PX with Auto Scaling Groups in AWS, you should use Secrets to have Kubernetes provision the ETCD certificates.  This way, the certificates will be provisioned on any new EC2 instance automatically.
 
-##### Installing behind the HTTP proxy
+Use `kubectl` to create the secret, for example:
+
+```
+# kubectl -n kube-system create secret generic etcd-certs --from-file=etcd-secrets/
```
Comment (harsh-px) [inline _includes/k8s-spec-generate.md]: We will need more specific examples on how to create this secret.. @piyush-nimbalkar if you have a working example, can you add more details. You should be able to commit directly to this branch.

File: _includes/k8s-spec-generate.md [hunk 5]
```diff
@@ -4,11 +4,40 @@ To generate the spec file for the 1.3 release, head on to [1.3 install page](htt
...
+      secretName: etcd-certs
+      items:
+      - key: pwx-ca.crt
+        path: pwx-ca.crt
+      - key: pwx-user-cert.crt
+        path: pwx-user-cert.crt
+      - key: pwx-user-key.key
+        path: pwx-user-key.key
+```
+
```
Comment (piyush-nimbalkar) [inline _includes/k8s-spec-generate.md]: we should mention that the options that we will be using in the container params `-key = /etc/pwx/etccerts/pwx-ca.crt` and so on. At least a reminder under what path the secrets will be found in the pod (/etc/pwx/etccerts)

File: _includes/k8s-spec-generate.md (hunk 4 above)
Comment (piyush-nimbalkar) [reply _includes/k8s-spec-generate.md]: okay, will do

File: _includes/k8s-spec-generate.md (hunk 5 above)
Comment (gourao) [reply _includes/k8s-spec-generate.md]: Can you update the PR with these edits?
